google_guest_agent/network/dhclient.go

```diff
@@ -0,0 +1,109 @@
+package network
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/GoogleCloudPlatform/guest-agent/google_guest_agent/run"
+	"github.com/GoogleCloudPlatform/guest-agent/metadata"
+	"github.com/GoogleCloudPlatform/guest-agent/utils"
+	"github.com/GoogleCloudPlatform/guest-logging-go/logger"
+)
+
+type dhclient struct{}
+
+func (n dhclient) IsManaging(ctx context.Context, iface string) (bool, error) {
+	// Check if the dhclient CLI exists.
+	result := run.WithOutput(ctx, "command", "-v", "dhclient")
+	if result.ExitCode == 1 {
+		// Failed to find dhclient command.
+		return false, nil
+	} else if result.ExitCode != 0 {
+		// Something went wrong when executing the command.
+		return false, fmt.Errorf("error getting dhclient command: %v", result.StdErr)
+	}
+
+	// Check for the process pertaining the dhclient managing the primary interface.
+	result = run.WithOutput(ctx, "/bin/sh", "-c", "ps aux | grep dhclient")
```

Comment:
Do you need to pass `/bin/sh -c`? I believe you should be able to run `ps` command directly

Reply:
This is so we can use the pipe to `grep` have the output be anything containing `dhclient`. This avoids having to loop through each line of the output and check for matches ourselves.